### PR TITLE
adde e2e tests for assets

### PR DESCRIPTION
**Pull request by aramikm**
# Description
Adds E2E tests for uploaded assets and requests referencing these assets

for e2e tests to work we need to run the redis and worker and ipfs in the background. Might need to create a docker image for it

# Verification

```bash

> content-publishing-service@0.1.0 test:e2e
> jest --config ./apps/api/test/jest-e2e.json --detectOpenHandles

 PASS  apps/api/test/app.e2e-spec.ts (16.807 s)
  AppController E2E request verification!
    ✓ (GET) /api/health (122 ms)
    Validate Route params
      ✓ invalid userDsnpId should fail (41 ms)
      ✓ invalid DsnpContentHashParam should fail (30 ms)
    (POST) /api/content/:dsnpUserId/broadcast
      ✓ valid request without uploaded assets should work! (31 ms)
      ✓ valid request with uploaded assets should work! (1825 ms)
      ✓ request with not uploaded assets should fail! (44 ms)
      ✓ empty body should fail (30 ms)
      ✓ empty content should fail (28 ms)
      ✓ empty published should fail (25 ms)
      ✓ invalid published should fail (24 ms)
      ✓ invalid assets type should fail (30 ms)
      ✓ image asset without references should fail (25 ms)
      ✓ image asset with non unique references id should fail (23 ms)
      ✓ link asset without href should fail (26 ms)
      ✓ link asset with invalid href protocol should fail (30 ms)
      ✓ hashtag without name should fail (24 ms)
      ✓ mentioned tag without mentionedId should fail (24 ms)
      ✓ invalid tag type should fail (25 ms)
      ✓ location with invalid units should fail (36 ms)
      ✓ location with empty name should fail (26 ms)
    (POST) /api/content/:dsnpUserId/reply
      ✓ valid request without assets should work! (33 ms)
      ✓ valid request with uploaded assets should work! (1654 ms)
      ✓ request with not uploaded assets should fail! (39 ms)
      ✓ empty body should fail (30 ms)
      ✓ empty inReplyTo should fail (29 ms)
      ✓ invalid inReplyTo should fail (32 ms)
    (POST) /api/content/:dsnpUserId/reaction
      ✓ valid request should work! (26 ms)
      ✓ valid emoji should fail (32 ms)
      ✓ valid apply amount should fail (24 ms)
      ✓ invalid inReplyTo should fail (24 ms)
    (PUT) /api/content/:dsnpUserId/:contentHash
      ✓ valid request without assets should work! (41 ms)
      ✓ valid request with uploaded assets should work! (1650 ms)
      ✓ request with not uploaded assets should fail! (39 ms)
      ✓ invalid targetAnnouncementType should fail (25 ms)
    (DELETE) /api/content/:dsnpUserId/contentHash
      ✓ valid request should work! (27 ms)
    (PUT) /api/profile/:userDsnpId
      ✓ valid request without assets should work! (35 ms)
      ✓ valid request with uploaded assets should work! (1658 ms)
      ✓ request with not uploaded icon should fail! (27 ms)
      ✓ request with non-image uploaded assets should fail! (1669 ms)
      ✓ empty profile should fail (45 ms)
      ✓ invalid published should fail (27 ms)
      ✓ non unique reference ids should fail (27 ms)
    (PUT) /api/asset/upload
      ✓ valid request should work! (33 ms)
      ✓ invalid mime should fail (25 ms)
      ✓ valid request should work! (2746 ms)
      ✓ upload asset should be uploaded to IPFS (2102 ms)

Test Suites: 1 passed, 1 total
Tests:       46 passed, 46 total
Snapshots:   0 total
Time:        16.854 s
Ran all test suites.
```

### CODE REVIEW

File: apps/worker/src/asset_processor/asset.processor.service.ts [hunk 1]
```diff
@@ -43,7 +43,7 @@ export class AssetProcessorService extends WorkerHost {
     const secondsPassed = Math.round((Date.now() - job.timestamp) / 1000);
     const expectedSecondsToExpire = 5 * 60; // TODO: get from config
     const secondsToExpire = Math.max(0, expectedSecondsToExpire - secondsPassed);
-    const result = await this.redis.pipeline().del(job.data.contentLocation).expire(job.data.metadataLocation, secondsToExpire, 'LT').exec();
+    const result = await this.redis.pipeline().expire(job.data.contentLocation, secondsToExpire, 'LT').expire(job.data.metadataLocation, secondsToExpire, 'LT').exec();
```
Comment (aramikm) [inline apps/worker/src/asset_processor/asset.processor.service.ts:46]: should expire isntead of delete to make sure a certain race condition will not happen

File: env.template [hunk 2]
```diff
@@ -12,3 +12,10 @@ HEALTH_CHECK_MAX_RETRY_INTERVAL_SECONDS=10
 HEALTH_CHECK_MAX_RETRIES=4
 CAPACITY_LIMIT='{"type":"percentage", "value":80}'
 
+IPFS_ENDPOINT="http://127.0.0.1:5001"
+IPFS_BASIC_AUTH_USER=""
+IPFS_BASIC_AUTH_SECRET=""
+IPFS_GATEWAY_URL="http://127.0.0.1:8080/ipfs/[CID]"
+
+# should be enabled for e2e tests
+# ENABLE_DEV_CONTROLLER=true
```
Comment (aramikm) [inline env.template:21]: question: how can we enforce a specific .env file for e2e tests?

Comment (saraswatpuneet) [reply env.template:21]: we can add one for e2e and in the script we can copy it to .env such that node can source it

**Review by saraswatpuneet — APPROVED** [review]
lgtm

**Pull request merged**